src/Engine/ProtoCore/ProcedureTable.cs

```diff
@@ -110,13 +142,27 @@ public override bool Equals(object obj)
 
     public class ProcedureTable
     {
-        public int runtimeIndex { get; private set; }
+        public int runtimeIndex { get; set; }
         public List<ProcedureNode> procList { get; set; }
         
         public ProcedureTable(int runtimeindex)
         {
             runtimeIndex = runtimeindex;
             procList = new List<ProcedureNode>();
+            foreach (ProcedureNode procNode in procList)
```

Comment:
What's the use of this `foreach` loop when `procList` is always going to be empty?